data/generic_vector.go

```diff
@@ -56,3 +56,13 @@ func (v *genVector) Type() FieldType {
 func (v *genVector) Extend(i int) {
 	(*v) = append((*v), make([]gen, i)...)
 }
+
+func (v *genVector) InsertAt(i int, val interface{}) {
+	if v.Len() < i {
```

Comment:
If you have a vector of length 3, and you insert at 3, append does seem reasonable.

However, with the same vector, inserting at say 10, I don't think appending makes sense. I think should panic in this case.

Else my concern is that consumers might be off by one(+) and have more trouble catching the problem.